packages/plugin-infra/src/utils/createInjectHooksRenderer.tsx

```diff
@@ -44,16 +45,20 @@ export function createInjectHooksRenderer<PluginDefinition extends Plugin.Shared
             .filter(pickInjectorHook)
             .map((plugin) => (
                 <PropsContext.Provider key={plugin.ID} value={props}>
-                    <SinglePluginWithinErrorBoundary key={plugin.ID} plugin={plugin} />
+                    <NestedShadowRootIsolation data-plugin={plugin.ID}>
+                        <SinglePluginWithinErrorBoundary key={plugin.ID} plugin={plugin} />
+                    </NestedShadowRootIsolation>
                 </PropsContext.Provider>
             ))
         return <>{all}</>
     }
     return memo(function PluginsInjectionHookRenderErrorBoundary(props: PropsType) {
         return (
-            <ErrorBoundary>
-                <PluginsInjectionHookRender {...props} />
-            </ErrorBoundary>
+            <span data-plugin-render="">
```

Comment:
Not necessary. This span is used to wrap all plugins because there're many plugins and it's a noise. With this span, we can fold all plugins at once.

This `data-plugin-render` is not used anywhere. It's for a better view in the F12 DOM view.